### DISCUSSION TITLE
How to add a trace characteristic when create a farfield report?

### DISCUSSION THREAD

**Discussion by shawnxuzhang** [ansys/pyaedt#4831] [Q&A] [answered]
In HFSS GUI, after create a farfield report, i can add some favor trace characteristics like 'xdb10Beamwidth', is there any method in pyaedt can do this?
![Snipaste_2024-06-19_15-18-17](https://github.com/ansys/pyaedt/assets/52066349/efe23943-1b9f-4705-88ec-b6e23e6e2374)

**Comment by Samuelopez-ansys** [comment, marked as answer]
@shawnxuzhang

Yes, add_trace_characteristics

https://aedt.docs.pyansys.com/version/0.9/API/_autosummary/pyaedt.modules.report_templates.EyeDiagram.add_trace_characteristics.html

For example:
```
report = hfss.post.reports_by_category.far_field("GainTotal", "Setup : LastAdaptive", "3D_Sphere")
report.primary_sweep = "Phi"
report.create()
report.add_trace_characteristics("xdb10Beamwidth", ["3"], ["Full"])
```
